### PR TITLE
ISPN-11472 fixes for Hot Rod Client docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11472 

needs backport to 10.1.x

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/config_examples/hotrod_java_client.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<dependency>
+  <groupId>org.infinispan</groupId>
+  <artifactId>infinispan-client-hotrod</artifactId>
+  <version>${version.infinispan}</version>
+  <scope>test</scope>
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/config_examples/hotrod_java_client.xml]: no `<scope>`

File: documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [id='download_java_client']
-= Downloading the Hot Rod Java Client
-{brandname} provides a reference Hot Rod Java client implementation.
+= Getting the Hot Rod Java Client
+{brandname} provides a Hot Rod Java client implementation.
 
 //Community
 ifndef::productized[]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc]: we no longer zip the java hot rod client. Users must use the maven coordinates.

Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc]: thanks @pruivo fixed

File: documentation/src/main/asciidoc/topics/config_examples/hotrod_java_client.xml (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/config_examples/hotrod_java_client.xml]: fixed @pruivo thanks

File: documentation/src/main/asciidoc/topics/code_examples/HotRodConnectionPools.java [hunk 3]
```diff
@@ -2,8 +2,12 @@
 clientBuilder
    .addServer()
     .host("127.0.0.1")
-    .port(11222);
-   .connectionPool() <1>
-    .maxActive(10) <2>
-    .minIdle(20); <3>
+    .port(11222)
+   //Configure client connection pools.
+   .connectionPool()
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/code_examples/HotRodConnectionPools.java]: These lines aren't aligned. Space is missing.
```suggestion
    //Configure client connection pools.
    .connectionPool()
```

File: documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc [hunk 4]
```diff
@@ -1,13 +1,9 @@
...
-endif::productized[]
+Add the `infinispan-client-hotrod` dependency to your `pom.xml` as follows:
 
-//Product
-ifdef::productized[]
-Visit the link:{download_url}[{brandname} downloads] page on the {portal} to download the Java client and find client implementations in other languages.
-endif::productized[]
+[source,xml]
+----
+include::dependencies_maven/hotrod_java_client.xml[]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc]: this path doesn't exist:
```
Unresolved directive in ../../topics/proc_download_hotrod_java_client.adoc - include::dependencies_maven/hotrod_java_client.xml[]
```

File: documentation/src/main/asciidoc/topics/proc_forcing_return_values.adoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+Method calls that modify entries can return previous values for keys, for
+example:
+
+[source,java]
+----
+V remove(Object key);
+V put(K key, V value);
+----
+
+To reduce the amount of data sent across the network, RemoteCache operations
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_forcing_return_values.adoc]:
```suggestion
To reduce the amount of data sent across the network, `RemoteCache` operations
```

File: documentation/src/main/asciidoc/topics/proc_forcing_return_values.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+always return null values.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_forcing_return_values.adoc]:
```suggestion
always return `null` values.
```

File: documentation/src/main/asciidoc/topics/ref_hotrod_listeners.adoc [hunk 7]
```diff
@@ -110,31 +110,32 @@ which statically filter out all entries except the one whose key is `1`.
...
 filter class implementation.
-4. Deploy the JAR file in the {brandname} Server.
-
-On top of that, the client listener needs to be linked with this cache event
-filter factory by adding the factory's name to the @ClientListener annotation:
-
+. Add the JAR file to the `server/lib` directory of your {brandname} server
+installation directory.
+. Link the client listener with this cache event filter factory by adding the
+factory name to the @ClientListener annotation:
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/ref_hotrod_listeners.adoc]:
```suggestion
factory name to the `@ClientListener` annotation:
```

File: documentation/src/main/asciidoc/topics/ref_hotrod_remote_cache.adoc [hunk 8]
```diff
@@ -48,7 +49,7 @@ have to copy the contents yourself. Also you can do this with `entrySet` and
 Set<K> keysCopy = remoteCache.keySet().stream().collect(Collectors.toSet());
 ----
 
-Please use extreme cautiong with this as a large number of keys can and will cause
+Please use extreme caution with this as a large number of keys can and will cause
 OutOfMemoryError in the client.
 
 [source,java]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/ref_hotrod_remote_cache.adoc]: this example can be removed IMO. it didn't add much to the docs.

File: documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc (hunk 4 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_download_hotrod_java_client.adoc]: oh, man! thanks for catching that @pruivo

File: documentation/src/main/asciidoc/topics/ref_hotrod_remote_cache.adoc (hunk 8 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/ref_hotrod_remote_cache.adoc]: thanks. I've removed it.

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**